### DISCUSSION TITLE
Adding GDK101 sensor

### DISCUSSION THREAD

**Discussion by Szewcson** [arendst/Tasmota#18356] [Q&A]
Hi,

I want to add FTlab GDK-101 Gamma Radiation Sensor Module. Could you point me where I should start to add a serial device, since most guides I can find is about i2c devices.

**Comment by barbudor** [comment]
There are a few serial sensors
You can search for sensors using the API `TasmotaSerial`

**Reply by sfromis** [reply]
Additionally, I can recommend using ESP32, where the Berry language provides for easier development, which can work quite well for interfacing with serial sensors.

**Comment by Szewcson** [comment]
Ok, I will continue here. This sensor have both i2c, and UART. I think I want add both interfaces in the future but I will start with i2c. I also found that there is an unmerged PR with that sensor. So what is the proper way to deal with it? Also the xsns number from that PR is taken, which I should assign? https://github.com/arendst/Tasmota/pull/6221

**Comment by Szewcson** [comment]
Can I use xsns 25?

**Comment by arendst** [comment]
Take 106 and focus on i2c only.

**Comment by barbudor** [comment]
The unmerged PR is incomplete as it makes call to a library that was not part of the PR
If you can find that library may be that old PR code can be usefull, it if is using the i2c interface. 
You can then just copy paste the code from the Github web into a new file

This being said, an old unmerged and incomplete PR could as much make you loose many time VS restarting from scratch

Also, as pointed by sfromis, using an ESP32 and Berry language could be easier for a Tasmota beginner than creating a C driver for the 1st time. Berry can support both i2c and serial and you don't need to consider a driver number.

**Reply by Szewcson** [reply]
https://github.com/arendst/Tasmota/pull/18390 PR is ready for review.
